internal/langserver/handlers/signature_help.go

```diff
@@ -0,0 +1,38 @@
+package handlers
+
+import (
+	"context"
+
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+)
+
+func (svc *service) SignatureHelp(ctx context.Context, params lsp.SignatureHelpParams) (*lsp.SignatureHelp, error) {
+	_, err := ilsp.ClientCapabilities(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Just double checking: Do we not need to gate certain parts of the response based on the client capabilities?

 - `documentationFormat` - we cannot provide markdown even if the client asks for it - so I suppose there's nothing to do there until/unless https://github.com/hashicorp/terraform-ls/issues/1212 is addressed
 - `activeParameterSupport` - I guess we'd have to send literal `0` if client doesn't support it, because the struct field is `int` (i.e. not optional), so that's a problem similar to https://github.com/hashicorp/terraform-ls/issues/1212 is it? Although we don't track it anywhere explicitly yet.
 - `labelOffsetSupport` - this would be dependent on https://github.com/hashicorp/terraform-ls/issues/1211 right?

Reply:
`documentationFormat` and `labelOffsetSupport` shouldn't be a problem. But I keep thinking about `activeParameterSupport`. 

We currently set `activeParameter` on [SignatureInformation](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#signatureInformation) and this field was added in `3.16.0`. So I might just set this on [SignatureHelp](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#signatureHelp) to instead to be save?

Reply:
> We currently set activeParameter on [SignatureInformation](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#signatureInformation) and this field was added in 3.16.0. So I might just set this on [SignatureHelp](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#signatureHelp) to instead to be save?

Aye, I didn't realise _that_ minor difference either.

I was more referring to the fact that `activeParameter` in both cases is optional and if we were strictly following the spec, we shouldn't be sending it to the client, unless it explicitly sent us `activeParameterSupport: true`. But again - this is in practice hard to achieve since the underlying type is now `uint32` and not `*uint32`, so we cannot omit that unless we reimplement the relevant struct(s).